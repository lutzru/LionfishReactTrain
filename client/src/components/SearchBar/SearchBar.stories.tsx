import { action } from '@storybook/addon-actions'
import { boolean, text, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import 'utils/ i18n'
import SearchBar, { SearchBarProps } from './SearchBar'

const stories = storiesOf('SearchBar', module)

const props: SearchBarProps = {
    onSearchTextChanged: action('onSearchTextChanged'),
    onFilterToggle: action('onFilterToggle'),
    searchText: 'test',
    filterStateOffen: true,
    filterStateInArbeit: false,
    filterStateFertig: true,
}

const props2: SearchBarProps = {
    onSearchTextChanged: action('onSearchTextChanged'),
    onFilterToggle: action('onFilterToggle'),
    searchText: '',
    filterStateOffen: false,
    filterStateInArbeit: false,
    filterStateFertig: false,
}

stories.addDecorator(withKnobs)
stories.add('default view (interactive)', () => (
    <SearchBar
        onSearchTextChanged={action('onSearchTextChanged')}
        onFilterToggle={action('onFilterToggle')}
        searchText={text('Searchtext', '')}
        filterStateOffen={boolean('offen', false)}
        filterStateInArbeit={boolean('in Arbeit', false)}
        filterStateFertig={boolean('fertig', false)}
    />
))
stories.add('empty view', () => <SearchBar {...props2} />)
