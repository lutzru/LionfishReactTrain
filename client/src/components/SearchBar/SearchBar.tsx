import { Box, Checkbox, FormControlLabel, Grid, TextField } from '@material-ui/core'
import * as React from 'react'

//
// Typisierung der Props per Interface
//
export interface SearchBarProps {
    setIsOffen: (filterState: boolean) => void
    setIsInArbeit: (filterState: boolean) => void
    setIsFertig: (filterState: boolean) => void
    setSearchTerm: (searchText: string) => void

    filterStateOffen: boolean
    filterStateInArbeit: boolean
    filterStateFertig: boolean
    searchTerm: string
}

// JSX.Element - typisierung des Returnwerts der Funktion
// (props: SearchBarProps) - typisierte Props

const SearchBar = (props: SearchBarProps): JSX.Element => {
    return (
        <Box padding={3}>
            <Grid container alignItems="center" spacing={4}>
                <Grid item xs={12} sm={3}>
                    <TextField
                        id="searchtext"
                        label="Artikelbez.:"
                        value={props.searchTerm}
                        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                            props.setSearchTerm(e.target.value)
                        }}
                    />
                </Grid>

                <Grid container item xs={12} sm={9}>
                    <Grid item xs={12} sm={4}>
                        <FormControlLabel
                            control={
                                <Checkbox
                                    name="Offen"
                                    color="primary"
                                    checked={props.filterStateOffen}
                                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                                        props.setIsOffen(!props.filterStateOffen)
                                    }}
                                />
                            }
                            label="offen"
                        />
                    </Grid>

                    <Grid item xs={12} sm={4}>
                        <FormControlLabel
                            control={
                                <Checkbox
                                    name="InArbeit"
                                    color="primary"
                                    checked={props.filterStateInArbeit}
                                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                                        props.setIsInArbeit(!props.filterStateInArbeit)
                                    }}
                                />
                            }
                            label="in Arbeit"
                        />
                    </Grid>

                    <Grid item xs={12} sm={4}>
                        <FormControlLabel
                            control={
                                <Checkbox
                                    name="Fertig"
                                    color="primary"
                                    checked={props.filterStateFertig}
                                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                                        props.setIsFertig(!props.filterStateFertig)
                                    }}
                                />
                            }
                            label="fertig"
                        />
                    </Grid>
                </Grid>
            </Grid>
        </Box>
    )
}

export default SearchBar
