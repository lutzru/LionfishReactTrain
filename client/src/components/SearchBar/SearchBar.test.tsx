import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import SearchBar, { SearchBarProps } from './SearchBar'

jest.mock('react-i18next', () => ({
    useTranslation: (): unknown => ({ t: (key: string): string => key }),
}))
jest.mock('react', () => ({
    // @ts-ignore
    ...jest.requireActual('react'),
    useState: jest.fn(),
}))

const props: SearchBarProps = {
    onSearchTextChanged: jest.fn(),

    setIsOffen: jest.fn(),
    setIsInArbeit: jest.fn(),
    setIsFertig: jest.fn(),

    searchText: 'SearchText',
    filterStateOffen: true,
    filterStateInArbeit: true,
    filterStateFertig: true,
}

//
//  reproduzierbarer Test, testet nur die geÃ¤nderte Komponente
//  eingefrorener manueller Test.  / Ordner _snapshots_ mit einchecken
//  u - erneuert den Snapshot
//

describe('SearchBar Component', () => {
    // beforeEach(() => {
    //     const setState = jest.fn()
    //     // @ts-ignore
    //     useStateMock.mockImplementation((state) => [state, setState])
    // })

    // it('should render correctly with react-test-renderer', () => {
    //     const component = renderer.create(<SearchBar {...props} />)

    //     expect(component).toMatchSnapshot()
    // })

    it('should render correctly with enzyme', () => {
        const component = shallow(<SearchBar {...props} />)

        expect(toJson(component)).toMatchSnapshot()
    })
    // it('should handle a click correctly', () => {
    //     const onClickMock = jest.fn()
    //     const component = shallow(<SearchBar onClick={onClickMock} />)

    //     component.find(Typography).simulate('click', { currentTarget: { value: 0 } })
    //     expect(onClickMock).toHaveBeenCalledTimes(1)
    // })
})
