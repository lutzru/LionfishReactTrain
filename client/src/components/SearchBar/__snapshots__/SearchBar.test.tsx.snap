// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component should render correctly with enzyme 1`] = `
<Styled(MuiBox)
  padding={3}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    spacing={4}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={3}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        id="searchtext"
        label="Artikelnr.:"
        onChange={[Function]}
        value="SearchText"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      sm={9}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={4}
        xs={12}
      >
        <WithStyles(ForwardRef(FormControlLabel))
          control={
            <ForwardRef(WithStyles)
              checked={true}
              color="primary"
              name="Offen"
              onChange={[Function]}
            />
          }
          label="offen"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={4}
        xs={12}
      >
        <WithStyles(ForwardRef(FormControlLabel))
          control={
            <ForwardRef(WithStyles)
              checked={true}
              color="primary"
              name="InArbeit"
              onChange={[Function]}
            />
          }
          label="in Arbeit"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={4}
        xs={12}
      >
        <WithStyles(ForwardRef(FormControlLabel))
          control={
            <ForwardRef(WithStyles)
              checked={true}
              color="primary"
              name="Fertig"
              onChange={[Function]}
            />
          }
          label="fertig"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Styled(MuiBox)>
`;
